### PR TITLE
Fix AutoConstructSpline GC Alloc

### PR DESCRIPTION
**Pull request by batuhandrsn**

### CODE REVIEW

**Comment by Leprawel** [comment]
I suggest you to provide a short description of what is the problem and how this change fixes it.

File: Plugins/BezierSolution/BezierSpline.cs [hunk 1]
```diff
@@ -75,6 +75,9 @@ public SplineAutoConstructMode autoConstructMode
 				}
 			}
 		}
+		[HideInInspector] private Vector3[] m_rhs;
```
Comment (yasirkula) [inline Plugins/BezierSolution/BezierSpline.cs:78]: Please put a blank line above.

Comment (yasirkula) [inline Plugins/BezierSolution/BezierSpline.cs:78]: I've used "m_" notation in properties' backing fields only, please rename this to autoConstructedSplineRhs.

File: Plugins/BezierSolution/BezierSpline.cs [hunk 2]
```diff
@@ -75,6 +75,9 @@ public SplineAutoConstructMode autoConstructMode
...
+		[HideInInspector] private Vector3[] m_controlPoints;
```
Comment (yasirkula) [inline Plugins/BezierSolution/BezierSpline.cs:79]: Please rename to autoConstructedSplineControlPoints.

**Comment by yasirkula** [comment]
Thank you for the suggestion! In this particular case, there's no need because I understood the problem.

File: Plugins/BezierSolution/BezierSpline.cs [hunk 3]
```diff
@@ -75,6 +75,9 @@ public SplineAutoConstructMode autoConstructMode
...
+		[HideInInspector] private float[] m_tmp;
```
Comment (yasirkula) [inline Plugins/BezierSolution/BezierSpline.cs:80]: Please rename to autoConstructedSplineTmp.

File: Plugins/BezierSolution/BezierSpline.cs [hunk 4]
```diff
@@ -1048,57 +1051,58 @@ public void AutoConstructSpline()
 				return;
 			}
 
-			Vector3[] rhs;
-			if( m_loop )
-				rhs = new Vector3[n + 1];
-			else
-				rhs = new Vector3[n];
+			int rhsCount = m_loop ? n + 1 : n;
```
Comment (yasirkula) [inline Plugins/BezierSolution/BezierSpline.cs:1054]: You can simply rename this to rhsLength and remove rhsLength's original declaration at line 1073.

File: Plugins/BezierSolution/BezierSpline.cs [hunk 5]
```diff
@@ -1048,57 +1051,58 @@ public void AutoConstructSpline()
...
 
 			// Get first control points
-			int rhsLength = rhs.Length;
-			Vector3[] controlPoints = new Vector3[rhsLength]; // Solution vector
-			float[] tmp = new float[rhsLength]; // Temp workspace
+			int rhsLength = m_rhs.Length;
+			if( m_controlPoints == null || rhsLength != m_controlPoints.Length )
+				m_controlPoints = new Vector3[rhsLength]; // Solution vector
+			if( m_tmp == null || rhsLength != m_tmp.Length )
+				m_tmp = new float[rhsLength]; // Temp workspace
```
Comment (yasirkula) [inline Plugins/BezierSolution/BezierSpline.cs:1077]: We can move these declarations to immediately below m_rhs's declaration.

**Review by yasirkula — CHANGES_REQUESTED** [review]

**Review by yasirkula — APPROVED** [review]
LGTM!

**Pull request merged**